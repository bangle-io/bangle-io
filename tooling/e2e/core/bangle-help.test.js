const { SELECTOR_TIMEOUT, sleep, url, newPage } = require('../helpers');

let page, destroyPage;
beforeEach(async () => {
  ({ page, destroyPage } = await newPage(browser));

  // await jestPuppeteer.resetPage();
  await page.goto(url, { waitUntil: 'networkidle2' });

  await page.evaluate(() => localStorage.clear());
});

afterEach(async () => {
  await destroyPage();
});

test('Landing page is correct', async () => {
  const handle = await page.waitForSelector('.bangle-editor', {
    timeout: SELECTOR_TIMEOUT,
  });
  await sleep(100);

  const result = await handle.evaluate((node) => node.innerText);
  expect(result).toMatchSnapshot();
});
