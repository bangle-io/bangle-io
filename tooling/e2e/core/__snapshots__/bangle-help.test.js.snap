// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page is correct 1`] = `
"Getting Started

Hello,

Here is a short guide to get you started.

Creating a workspace

Workspaces are collection of notes that you can open with Bangle.io. You can choose to have multiple workspaces or keep all your notes in one workspace.

‚ö†Ô∏è Don't forget to backup before opening your data. If you are in hurry simply copy and paste all your notes to another location to create a backup.

To start taking notes, let us first create a new Workspace.




Click on the Action Palette icon at top right.

Run the New workspace action as shown below:




Once executed select one of the following options:

Your computer: This option allows bangle to open a directory in your computer, ideally have markdown notes. Note: It is currently only available in Chrome and Edge browsers.

Browser: If you cannot use the above option you can save your notes in the browser storage. Though it has the drawback of not opening any of your existing notes.

üí°We recommend that you store the notes in your computer to fully take advantage of all the features.

Next you will be shown a directory picker.

If you already have Markdown notes select that folder.

Or create a new directory that Bangle will use to save your notes.

Creating a note

There are two ways to create a new note:

Using an action: Execute the action New note from the Action Palette.

Inline backlink: In an existing note create a backlink to a new note by typing[[ followed by the name of your note you wish to create.

Using inline palette

You can type / anywhere in your document or press Command-/ (mac) Ctrl-/ (win) to bring a handy toolbar for inserting things.




Formatting text

Bangle uses the following markdown shortcuts to format text:

bold Cmd+B

italic Cmd+I

For link Cmd+K or select text and paste link on it.

~strikethrough~

For image drag or paste it directly.

\`\`\`for code block

\`inline code\`

* followed by space for bullet point or Cmd-shift-8

1. follow by space for numbered list or Cmd-shift-9

[ ] followed by space for todo list or Cmd-shift-7

You can also select text to bring up a floating menu to making formatting changes.


"
`;
